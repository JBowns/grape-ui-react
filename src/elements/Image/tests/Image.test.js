import React from 'react';
import 'jest-styled-components';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { ThemeProvider } from 'styled-components';
import { Image } from '..';

configure({ adapter: new Adapter() });

describe('Image Component base', () => {
  it('should return a shallow Image object with base styling', () => {
    const component = mount(<ThemeProvider theme={{}}><Image alt="blue flower" src="https://images.pexels.com/photos/67636/rose-blue-flower-rose-blooms-67636.jpeg" /></ThemeProvider>);
    expect(toJson(component.find('Img'))).toMatchSnapshot();
  });
  it('should return a Image object with a console error', () => {
    const component = mount(<ThemeProvider theme={{}}><Image src="https://images.pexels.com/photos/67636/rose-blue-flower-rose-blooms-67636.jpeg" /></ThemeProvider>);
    expect(toJson(component.find('Img'))).toMatchSnapshot();
  });
  it('should return a Image object with maxWidth applied', () => {
    const component = mount(<ThemeProvider theme={{}}><Image alt="blue flower" maxWidth="200px" src="https://images.pexels.com/photos/67636/rose-blue-flower-rose-blooms-67636.jpeg" /></ThemeProvider>);
    expect(toJson(component.find('Img'))).toMatchSnapshot();
  });
  it('should return a Image object with width applied', () => {
    const component = mount(<ThemeProvider theme={{}}><Image alt="blue flower" src="https://images.pexels.com/photos/67636/rose-blue-flower-rose-blooms-67636.jpeg" width={[1, 3 / 4, 2 / 3, 1 / 2]} /></ThemeProvider>);
    expect(toJson(component.find('Img'))).toMatchSnapshot();
  });
});
