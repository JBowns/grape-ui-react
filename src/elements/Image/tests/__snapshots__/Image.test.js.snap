// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Component base should return a Image object with maxWidth applied 1`] = `
.c0 {
  width: 100%;
  max-width: 200px;
}

<Img
  alt="blue flower"
  className="c0"
  container={[Function]}
  decode={true}
  loader={false}
  src="https://images.pexels.com/photos/67636/rose-blue-flower-rose-blooms-67636.jpeg"
  unloader={false}
/>
`;

exports[`Image Component base should return a Image object with width applied 1`] = `
.c0 {
  width: 100%;
  max-width: 100%;
}

@media screen and (min-width:40em) {
  .c0 {
    width: 75%;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    width: 66.66666666666666%;
  }
}

@media screen and (min-width:64em) {
  .c0 {
    width: 50%;
  }
}

<Img
  alt="blue flower"
  className="c0"
  container={[Function]}
  decode={true}
  loader={false}
  src="https://images.pexels.com/photos/67636/rose-blue-flower-rose-blooms-67636.jpeg"
  unloader={false}
/>
`;

exports[`Image Component base should return a Image object without alt base styling 1`] = `
.c0 {
  width: 100%;
  max-width: 100%;
}

<Img
  alt=""
  className="c0"
  container={[Function]}
  decode={true}
  loader={false}
  src="https://images.pexels.com/photos/67636/rose-blue-flower-rose-blooms-67636.jpeg"
  unloader={false}
/>
`;

exports[`Image Component base should return a shallow Image object with base styling 1`] = `
.c0 {
  width: 100%;
  max-width: 100%;
}

<Img
  alt="blue flower"
  className="c0"
  container={[Function]}
  decode={true}
  loader={false}
  src="https://images.pexels.com/photos/67636/rose-blue-flower-rose-blooms-67636.jpeg"
  unloader={false}
/>
`;
